src/engine/QueryPlanner.cpp

```diff
@@ -1401,6 +1404,14 @@ void QueryPlanner::applyFiltersIfPossible(
   row.insert(row.end(), addedPlans.begin(), addedPlans.end());
 }
 
+template void QueryPlanner::applyFiltersIfPossible<true>(
+    vector<QueryPlanner::SubtreePlan>& row,
+    const vector<SparqlFilter>& filters) const;
+
+template void QueryPlanner::applyFiltersIfPossible<false>(
+    vector<QueryPlanner::SubtreePlan>& row,
+    const vector<SparqlFilter>& filters) const;
```

Comment:
If they are private and only called from within the QueryPlanner.cpp you don't need the explicit instantiations.